verify.go

```diff
@@ -39,3 +43,95 @@ func VerifyInTotoEnvelope(ctx context.Context, env *dsse.Envelope, provider clie
 
 	return stmt, nil
 }
+
+func VerifyInTotoEnvelopeExt(ctx context.Context, env *Envelope, provider client.OpenIdProvider) (*intoto.Statement, error) {
+	tl := GetTL(ctx)
+
+	// enforce payload type
+	if env.PayloadType != intoto.PayloadType {
+		return nil, fmt.Errorf("unsupported payload type %s", env.PayloadType)
+	}
+
+	// verify signatures and transparency log entry
+	verifier := NewOPKSignerVerifier(provider)
+	for _, sig := range env.Signatures {
+		if sig.Extension.Kind != OpkSignatureID {
+			return nil, fmt.Errorf("error unsupported signature kind: %s", sig.Extension.Kind)
+		}
+
+		// verify opk signature
+		payload, err := base64.StdEncoding.Strict().DecodeString(env.Payload)
+		if err != nil {
+			return nil, fmt.Errorf("error failed to decode OPK payload: %w", err)
+		}
+		encPayload := dsse.PAE(intoto.PayloadType, payload)
+		pkToken := sig.Extension.Ext["pkt"].([]byte)
```

Comment:
We should check this case works to avoid a panic on an invalid type, something like:

```suggestion
		pkToken, ok := sig.Extension.Ext["pkt"].([]byte)
		if !ok {
			return nil, fmt.Errorf("expected pkt to be of type []byte, got %T", sig.Extension.Ext["pkt"])
		}
```

We might also want to do the same on the access to check the field exists.

(Likewise for `sig.Extension.Ext["tl"]` below)

Reply:
fixed [64ddde5](https://github.com/openpubkey/signed-attestation/pull/4/commits/64ddde5ad993413507ad27f100ca2327ddaab2cb)